### DISCUSSION TITLE
JBD BMS blocking charger from charging

### DISCUSSION THREAD

**Discussion by morty41** [Louisvdw/dbus-serialbattery#507] [Q&A]
TL;DR JBD BMS blocking charger from charging

Setting up a new system with 3,7v Lith Ion 18650 charged via Victron MPPT
I have a  
Victron 30/100 MPPT
JBD BMS 60a
DIY battery 18650 lithium ion 7x3.7v 180ah 24v
RPI Venus OS large v2.9 with node red enabled
SerialBattery 0.14.3

I have changed Victron 100/30 MPPT settings using the Victron Connect 
29.05v Absorption 
29.05V Float
29.05v Equalization

I have changed the SerialBattery utils.py to

utils.py
MIN_CELL_VOLTAGE = 3.6
MAX_CELL_VOLTAGE = 4.1
FLOAT_CELL_VOLTAGE = 4.1
MAX_VOLTAGE_TIME_SEC = 15*60
SOC_LEVEL_TO_RESET_VOLTAGE_LIMIT = 90

I have set the JBD BMS to 

FullChargeVol             4150
ChargeEndVol              3000
BalanceStartVoltage       4000

In Remote Console 
DVCC set

SShot2
![SShot2](https://user-images.githubusercontent.com/78052454/229351903-dc56df8e-1324-427a-8f73-3fbdf6df0190.png)

and REBOOTed 

But there is something not allowing it to charge. 

Victron Connect shows no reason, only showing

Status ON

When i look at the Remote Console Devicelist>SerialBattery JBD> IO

SShot1
![SShot1](https://user-images.githubusercontent.com/78052454/229351884-ab6fa6de-aab9-4a72-947f-40d125bcff2a.png)

> Parameters

SShot3
![SShot3](https://user-images.githubusercontent.com/78052454/229351938-efd93413-bc21-41b7-b1a2-24b1cf69c217.png)

It shows the Allow to charge set to NO and ChargeCurrentLimit to 0.00A

If I disconnect the JBD BMS from the RPI it all charges up nicely.  Its been taxing me on an off for over a week now so any ideas as to where i have gone wrong  or what i could try would be gratefully appreciated.

**Comment by morty41** [comment]
I have since changed the CCCM and DCCM in utils.py and the charge controller is now allowing charge. 

######### Cell Voltage limitation #########

# Charge current control management referring to cell-voltage enable (True/Fals>
CCCM_CV_ENABLE = False
# Discharge current control management referring to cell-voltage enable (True/F>
DCCM_CV_ENABLE = False
